### PR TITLE
Added some crash workarounds

### PR DESCRIPTION
**Pull request by WizzardMaker**
Added some log stuff to let the user know that the game didnt crash (for now)

Added a fix for the startup crash after logging into XBox
Added a "lets hope it works" patch for every exception that have to do with store/xbox/etc.

### CODE REVIEW

File: host/src/patches/MurderScene.cpp [hunk 1]
```diff
@@ -41,8 +41,8 @@ struct TpwStubCtx
 };
 
 PTP_WORK(WINAPI* g_origCreateThreadpoolWork)(_In_        PTP_WORK_CALLBACK    pfnwk,
-											 _Inout_opt_ PVOID                pv,
-											 _In_opt_    PTP_CALLBACK_ENVIRON pcbe);
+	_Inout_opt_ PVOID                pv,
```
Comment (nta) [inline host/src/patches/MurderScene.cpp]: Indentation changes are out of scope for a PR.

File: host/src/patches/MurderScene.cpp [hunk 2]
```diff
@@ -61,8 +61,8 @@ VOID WINAPI TpwStub(PTP_CALLBACK_INSTANCE cb, LPVOID argument, PTP_WORK work)
 }
 
 PTP_WORK WINAPI CreateThreadpoolWork_Custom(_In_        PTP_WORK_CALLBACK    pfnwk,
-											_Inout_opt_ PVOID                pv,
-											_In_opt_    PTP_CALLBACK_ENVIRON pcbe)
+	_Inout_opt_ PVOID                pv,
```
Comment (nta) [inline host/src/patches/MurderScene.cpp]: Same as above.

File: host/src/patches/MurderScene.cpp [hunk 3]
```diff
@@ -75,23 +75,49 @@ LONG WINAPI MurderSceneVEH(PEXCEPTION_POINTERS ptrs)
 		bool found = false;
 
 		{
+			printf("Warning: Exception catched! Trying to prevent...\n");
```
Comment (nta) [inline host/src/patches/MurderScene.cpp]: There's no console attached at all - how is this going to output anything? Also - typo...

File: host/src/patches/MurderScene.cpp [hunk 4]
```diff
@@ -75,23 +75,49 @@ LONG WINAPI MurderSceneVEH(PEXCEPTION_POINTERS ptrs)
...
 			{
+				printf("Info: Loading Exception. Preventing Crash...\n");
+				ptrs->ContextRecord->Rip = (DWORD64)RaiseTrailBlazer;
+				ptrs->ContextRecord->Rsp &= 0xFFFFFFFFFFFFFFF0;
+				ptrs->ContextRecord->Rsp |= 8;
+
+				return EXCEPTION_CONTINUE_EXECUTION;
+			}
+			else {
+				printf("Warning: Unknown Exception! Experimental workaround...\n");
```
Comment (nta) [inline host/src/patches/MurderScene.cpp]: Does this one fix the 'crash after loading'? It could be base addresses differ on some systems.

File: host/src/patches/MurderScene.cpp [hunk 5]
```diff
@@ -75,23 +75,49 @@ LONG WINAPI MurderSceneVEH(PEXCEPTION_POINTERS ptrs)
...
+
 				ptrs->ContextRecord->Rip = (DWORD64)RaiseTrailBlazer;
 				ptrs->ContextRecord->Rsp &= 0xFFFFFFFFFFFFFFF0;
 				ptrs->ContextRecord->Rsp |= 8;
 
 				return EXCEPTION_CONTINUE_EXECUTION;
 			}
 		}
+		else {
+			if (((uintptr_t)ptrs->ExceptionRecord->ExceptionAddress >> 36) == 0x7FF)
```
Comment (nta) [inline host/src/patches/MurderScene.cpp]: Egh, no, just no. Also - code duplication?!

File: host/src/patches/MurderScene.cpp [hunk 6]
```diff
@@ -103,4 +129,4 @@ void InitializeMurderScene()
 	MH_EnableHook(MH_ALL_HOOKS);
 
 	AddVectoredExceptionHandler(0, MurderSceneVEH);
-}
\ No newline at end of file
+}
```
Comment (nta) [inline host/src/patches/MurderScene.cpp:114]: More whitespace changes. ew.

**Review by nta — CHANGES_REQUESTED** [review]
Shoddy work.

File: host/src/patches/MurderScene.cpp (hunk 4 above)
Comment (WizzardMaker) [reply host/src/patches/MurderScene.cpp]: Yes that fixes it
